src/routes/api/sections/bookSection.ts

```diff
@@ -0,0 +1,176 @@
+import { Request, Response } from 'express';
+import timeBlockModel from '../../../models/timeBlock.model';
+import sectionModel from '../../../models/section.model';
+import Room from '../../../models/room.model';
+import { DateTime } from 'luxon';
+
+const bookSectionRoute = async (req: Request, res: Response): Promise<void> => {
+    const id = req.params.id;
+    const { emails, startsAt, endsAt } = req.body;
+
+    // Checks for needed request body variables
+    if (!id || !emails || !startsAt || !endsAt) {
+        res.status(400);
+        return;
+    }
+
+    // Retrieves section with the section id given
+    const sectionInformation = await sectionModel.findOne({ _id: id }).catch((error) => {
+        res.status(500).json({ error });
+        return;
+    });
+
+    if (!sectionInformation) {
+        res.status(404);
+        return;
+    }
+
+    // Retrieves room that the section corresponds to
+    const room = await Room.findOne({ _id: sectionInformation.roomId }).catch((error) => {
+        res.status(404).json({ error });
+        return;
+    });
+
+    if (!room) {
+        res.status(404);
+        return;
+    }
+
+    const start = DateTime.fromISO(String(startsAt), { zone: 'America/Toronto' });
+    const end = DateTime.fromISO(String(endsAt), { zone: 'America/Toronto' });
+
+    // Determine if the requested booking is valid by checking if all of the hours fall under open times and that the section will never be over capacity
+    let currHourStart = start;
+    let currHourEnd = currHourStart.plus({ hours: 1 });
+    let invalidHours = false;
+    let overCapacity = false;
+    while (currHourStart < end) {
+        // Finds schedule start and end for the day that the current hour falls on
+        const currWeekDay = currHourStart.weekday;
+        let scheduleDayStart = 0;
+        let scheduleDayEnd = 0;
+        for (const day of room.schedule) {
+            if (day.dayOfWeek == currWeekDay) {
+                scheduleDayStart = day.start;
+                scheduleDayEnd = day.end;
+            }
+        }
+
+        // Checks if the hour falls under an open time for the room and if it does adds hour to the list of available times
+        if (room.closed || currHourStart.weekday != currHourEnd.weekday || currHourStart.hour < scheduleDayStart || currHourEnd.hour > scheduleDayEnd) {
```

Comment:
It also might be good to see if the difference between `start` and `end` is exactly a number of hours (ex: 4 hours instead of 3 hours 46 minutes 5 seconds) and falls on the start of an hour (ex: starts at 9:00 AM instead of 9:16 AM)